ios/Example/Tests/TrifleCertificateTests.swift

```diff
@@ -9,49 +9,49 @@ import Trifle
-        
+    func testVerifyTrifleCertificateWithRoot_succeeds() throws {
+        let deviceCertificate = try TrifleCertificate(data: TestFixtures.deviceTrifleCertEncoded3!)
+        let rootCertificate = try TrifleCertificate(data: TestFixtures.rootTrifleCertEncoded3!)
+
         let isVerified = try deviceCertificate.verify(
-            certificateRequest: mobileCertReq,
-            intermediateChain: [],
-            rootCertificate: rootCertificate
+            certificateRequest: nil,
+            intermediateTrifleChain: [],
+            rootTrifleCertificate: rootCertificate
         )
         
         XCTAssertTrue(isVerified)
     }
     
-    func testVerifyCertificate_fails() throws {
-        let trifle = try Trifle(reverseDomain: TestFixtures.reverseDomain)
-        let keyHandle = try trifle.generateKeyHandle()
-        
-        let mobileCertReq = try trifle.generateMobileCertificateRequest(keyHandle: keyHandle)
-        
-        let deviceCertificate = Certificate(version: 0, certificate: TestFixtures.deviceCertEncoded)
-        let otherRootCertificate = Certificate(version: 0, certificate: TestFixtures.otherRootCertEncoded)
+    func testVerifyTrifleCertificateWithIntermediate_succeeds() throws {
+        let deviceCertificate = try TrifleCertificate(data: TestFixtures.deviceTrifleCertEncoded3!)
+        let rootCertificate = try TrifleCertificate(data: TestFixtures.rootTrifleCertEncoded3!)
+
+        let isVerified = try deviceCertificate.verify(
+            certificateRequest: nil,
+            intermediateTrifleChain: [rootCertificate],
+            rootTrifleCertificate: nil
+        )
         
-        XCTAssertThrowsError(try deviceCertificate.verify(
-            certificateRequest: mobileCertReq,
-            intermediateChain: [],
-            rootCertificate: otherRootCertificate ),
-                             "Certificate is invalid.")
+        XCTAssertTrue(isVerified)
     }
     
-    func testExpiredCertificate() throws {
-                
-        let validCertificate = Certificate(version: 0, certificate: TestFixtures.validCertEncoded)
-        let expiredCertificate = Certificate(version: 0, certificate: TestFixtures.expiredCertEncoded)
+    func testSerializeTrifleCertificate_succeeds() throws {
+        let deviceCertificate = try TrifleCertificate(data: TestFixtures.deviceTrifleCertEncoded3!)
 
-        XCTAssertTrue(try validCertificate.verify(intermediateChain: Array<Certificate>()))
-    
-        XCTAssertThrowsError(try expiredCertificate.verify(intermediateChain: Array<Certificate>()),
-                             "Certificate is expired.")
+        // serialize
+        let encoder = JSONEncoder()
+        let jsonData = try encoder.encode(deviceCertificate)
+
+        // de-serialized
+        let decoder = JSONDecoder()
+        let decoded = try decoder.decode(TrifleCertificate.self, from: jsonData)
+        
+        // verify the decoded certificate
```

Comment:
For a round trip test of de/serializing, we should check the equality of the object rather than testing its api, i.e. #verify